src/Client.php

```diff
@@ -20,34 +24,36 @@ class Client
     protected $endpointUrl;
 
     /**
-     * @var array
-     */
-    protected $authorizationHeaders;
-
-    /**
-     * @var \GuzzleHttp\Client
+     * @var ClientInterface
      */
     protected $httpClient;
 
     /**
      * @var array
      */
-    protected $httpOptions;
-
+    protected $httpHeaders;
 
     /**
      * Client constructor.
      *
      * @param string $endpointUrl
      * @param array $authorizationHeaders
      * @param array $httpOptions
+     * @param ClientInterface $httpClient
      */
-    public function __construct(string $endpointUrl, array $authorizationHeaders = [], array $httpOptions = [])
+    public function __construct(string $endpointUrl, array $authorizationHeaders = [], array $httpOptions = [], ClientInterface $httpClient = null)
     {
+        $headers = array_merge(
+            $authorizationHeaders,
+            $httpOptions['headers'] ?? [],
+            ['Content-Type' => 'application/json']
+        );
+
+        unset($httpOptions['headers']);
```

Comment:
I actually do have a little request here. If you can add an inlined comment explaining why you're unsetting the headers before using them to construct the client that'd be great!
It makes sense now that you explained it to me, but I know that to someone else it might not make much sense, even I can forget the rationale behind this a few weeks later